Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -0,0 +1,334 @@
+
+import UIKit
+import MapboxDirections
+import MapboxCoreNavigation
+import MapboxMaps
+import Turf
+import CoreLocation
+
+/// A components, designed to help manage `NavigationMapView` ornaments logic.
+class OrnamentsController: NavigationComponent, NavigationComponentDelegate {
```

Comment:
It would be problematic to nest it inside `NavigationMapView` since this class also closely operates with `NavigationView` members.

Reply:
It would just be syntactic sugar: the type would be nested, but that doesn’t change the relationship between NavigationMapView and this class in terms of object pointers. We’re doing something similar with electronic horizon types.

Regardless, I think “map view” should be somewhere in the type name, since there are other kinds of ornaments, and “ornament” should be singular.

```swift
extension NavigationMapView {
    class OrnamentController: NavigationComponent, NavigationComponentDelegate {
        // All this remains unchanged.
    }
}
```